src/hardware/vga_draw.cpp

```diff
@@ -2161,6 +2164,13 @@ void VGA_SetupDrawing(uint32_t /*val*/)
 	    (vga.draw.bpp != bpp) || fps_changed) {
 		VGA_KillDrawing();
 
+		if (width > SCALER_MAXWIDTH || height > SCALER_MAXHEIGHT) {
```

Comment:
Maybe it would be better to print a warning log and simply clip the values to the maximum? Terminating the emulation seems rather drastic to me.

Reply:
I agree. I think asserts are fine, but in general we should be as lenient as possible, and err on the side of "not crashing" whenever we can. Logging a warning and clipping would be better, I think.

Reply:
Yup - this is really the equivalent to "out of range" error:

![images](https://github.com/dosbox-staging/dosbox-staging/assets/1557255/ca97b922-a9e4-4974-b2f0-f7421d591492)

That's better - Will clamp it and log strongly.

But if I honestly expected even bigger resolutions beyond this - then absolutely: let's just accommodate them here too; but I want the data to lead me to that value instead of adding random contingency.

I'll try to work backward through the tweaked resolution calculations and see if there's a limit.

Reply:
Updated to log an error and limit the width to 1630 adapter pixels wide.

Going back through the registers, setting the overscan bit (b1) combined with setting all eight bits in the Horizontal Display End Register will produce a horizontal resolution of 4600 pixels (`1 + (255 | (0b10 << 7)) * 9`), which in turn will produce a very slow refresh rate of ~5 Hz (if the vertical is still 1200 lines). 

Well, I guess we'll wait for reports of needing more, if such software exists!

Reply:
Yeah, log a warning and let it crash/misbehave in the emulation but prevent DOSBox from crashing should be our philosophy 👍🏻 We'll still get report, just not random crash report 😄